File: Java/Exercism/CollatzConjecture.java
class CollatzCalculator {

    int computeStepCount(int start) {

        int howManySteps = 0;
        
        if (start <= 0){
            throw new IllegalArgumentException("Only positive integers are allowed");
        }

        if (start == 1){
            howManySteps = 0;
        } else {
            do {
                if (start % 2 == 0){
                    start = start / 2;
                    howManySteps++;
                } else {
                    start = (start * 3) + 1;
                    howManySteps++;
                }
            }
            while (start > 1);    
        }
    return howManySteps;
    }
}

File: JavaScript/Exercism/CollatzConjecture.js
//
// This is only a SKELETON file for the 'Collatz Conjecture' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const steps = num => {

    let numberSoFar = num;
    let steppySteps = 0;
    
    if (numberSoFar < 1) {
      throw new Error('Only positive integers are allowed');
    }
    
    while (numberSoFar > 1){
      if (numberSoFar % 2 == 0){
        numberSoFar /= 2;
      } else {
        numberSoFar = numberSoFar * 3 + 1;
      }
      steppySteps += 1;
      }
    
    return steppySteps;
  };